xbmc/music/MusicDatabase.cpp

```diff
@@ -593,11 +604,15 @@ bool CMusicDatabase::UpdateAlbum(CAlbum& album, bool OverrideTagData /* = true*/
           artistCredit->GetMusicBrainzArtistID());
         AddSongArtist(artistCredit->idArtist,
           song->idSong,
+          ROLE_ARTIST,
           artistCredit->GetArtist(),
-          artistCredit->GetJoinPhrase(),
-          artistCredit == song->artistCredits.begin() ? false : true,
           std::distance(song->artistCredits.begin(), artistCredit));
       }
+      // Having added artist credits (maybe with MBID) add the other contributing artists (MBID unknown)
+      for (VECMUSICROLES::const_iterator credit = song->GetContributors().begin(); credit != song->GetContributors().end(); ++credit)
```

Comment:
you're looping over these in two places, might be better to add an overloaded AddSongContributor that takes a vector and hide the looping in there